.github/workflows/ci.yaml

```
@@ -210,7 +210,32 @@ jobs:
           echo "${VERSION}"
           make build-${{ matrix.dist }}
 
-  ### e2e tests ###
+  build-gpu-operator-bundle-image:
+    if: github.ref == 'refs/heads/main' # Runs only if the event is from a main branch commit
+    runs-on: ubuntu-latest
```

We only want the OLM bundle to be built and pushed after the e2e tests pass as it is going to be consumed by external teams
```suggestion
    needs: [e2e-tests-containerd, e2e-tests-nvidiadriver]
    runs-on: ubuntu-latest
```